packages/coil-client/src/graphQlClient.ts

```diff
@@ -48,23 +52,57 @@ export class GraphQlClient {
     @inject(GraphQlClientOptions)
     private config: GraphQlClientOptions
   ) {
-    this.fetch = this.config.fetch
+    this.inBrowser = typeof window !== 'undefined'
+    this.fetch = this.inBrowser
+      ? this.config.fetch.bind(window)
+      : this.config.fetch
+  }
+
+  public async authenticate(email: string, password: string): Promise<void> {
+    const response = await this.fetch(
+      `${this.config.coilDomain}/api/auth/signin`,
+      {
+        method: 'POST',
+        credentials: 'same-origin',
+        headers: {
+          apiBasePath: this.config.coilDomain,
+          rid: 'thirdpartyemailpassword',
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          formFields: [
+            {
+              id: 'email',
+              value: email
+            },
+            {
+              id: 'password',
+              value: password
+            }
+          ]
+        })
+      }
+    )
+    const data = await response.json()
```

Comment:
there is `response.ok`
generally good to check before looking at the body (.json() )